### ISSUE TITLE
A doubt for a advice in 'Quick Start'

### ISSUE THREAD

**Issue by yutingzhao1991** [rackt/react-redux#242]
In [quick start](https://github.com/rackt/react-redux/blob/master/docs/quick-start.md#quick-start)

>It is advisable that only top-level components of your app (such as route handlers, for example) are aware of Redux. Components below them should be presentational and receive all data via props.

1: If I have a common component(like a selector), It need some data from store. If I follow this advice, will lead to this: everywhere I want to ues this component need to get the data from store and pass to the component with props.

2: If the component's position is deeply, this advice make the code terrible.

I think sometimes this advice is not the best practice, it dislead me when I first started to learn redux.

**Comment by gaearon** [comment]
I agree and we're slowly changing it: https://github.com/rackt/redux/pull/1186
Feel free to help out fleshing out those new docs!

My current stance on this is reflected in the last ten videos of my Egghead tutorial: https://egghead.io/series/getting-started-with-redux

In short, try to create presentational components first, but when you start passing unrelated props through intermediate components, it's time to extract a container component.

**Comment by emonidi** [comment]
It think what we're seeing here is a clash of perceptions of how an application data flow should be structured. I guess its present in all frameworks. I call it "Flat vs. Nested". I believe it is the creators preference to think of a components based application as a tree like structure and that is why he and many more think that there should be one single entry point of data to the presentation layer and all of the data should propagate down to the particular component that needs it. This however may create some architectural and performance problems (never did a research so this may be incorrect). For simple web application the "Nested model" is clear and simple to understand. You have a collection of people and you need to visually represent every each person and then it is easy to think about an UL component that gets all the  data and gives a piece of it of each LI component in it, and every IMG component takes the colllection.item.image_url bit and displays it.

However the creation of FLUX pattern I believe was made of totally different necessity.Let us imagine the facebook page. It is all one page. At this point there is very little part of the content of facebook that is happening out side the main container. You've got yourself, pages, friends, friends of  friends, videos, similar videos, likes, shares, chat, sharing and privacy preferences. Its a huge messy graph which needs to be displayed all in one main container. I hope facebook engineers get good salaries because they have very difficult problems to solve. I believe that the problems that an application like facebook is facing is the motivation for FLUX existence. If you take a close look of a FLUX application it may probably resemble a graph. Lots of nodes (stores and views) and connections between them (actions and dispatchers). Now the genius part of it is that they separated the connections between the nodes to actions and dispatchers so they become uni-directional, so representation/perception could be simplified. 

So long story short.... In my humble opinion, if your application's data representation looks like a graph use a the classical FLUX implementation. Else, if it is easier to think of your data as tree and you want to simplify things Redux may be a good and less painful solution with lots of goodies for free :)

**Comment by gaearon** [comment]
We no longer discourage creating container components in the updated docs.
http://redux.js.org/docs/basics/UsageWithReact.html
